extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/context/TransactionContext.java

```diff
@@ -32,6 +33,7 @@
 
     private final TransactionSynchronizationRegistry transactionSynchronizationRegistry = new TransactionSynchronizationRegistryImple();
     private final TransactionManager transactionManager = com.arjuna.ats.jta.TransactionManager.transactionManager();
+    private final Map<Transaction, TransactionScopeCleanup<?>> transactions = new HashMap<>();
```

Comment:
I'd prefer this to use a `ConcurrentMap`, so to not need `synchronized` on it further below.